both/access_control.js

```diff
@@ -258,38 +257,24 @@ if(Meteor.isServer) {
         'updatedAt',
         'createdAt',
       ];
-      return _onlyAllowedFields(fields, allowedFields);
+      return onlyAllowedFields(fields, allowedFields);
     },
     fetch: [ 'competitionId' ],
   });
 
   Registrations.allow({
     insert: function(userId, registration) {
-      return Registrations.userIsAllowed(userId, registration);
+      return Registrations.allowOperation(userId, registration, []);
```

Comment:
You could also sidestep the whole problem by reading the User fields from the Users collection when needed.

That would mean people can change things during the competition. That would be good if you misspell your name, but could also cause the problems you mention. I can't judge how bad that would be.

You could imagine many other schemes. These fields could get batch populated when registration closes, for example.